docs/source/yelpsoa_configs.rst

```diff
@@ -157,6 +157,169 @@ Marathon, Chronos, Tron, or ``paasta remote-run``.
+    Defaults to ``8888``.
+
+  * ``bounce_method``: Controls the bounce method; see `bounce_lib <generated/paasta_tools.bounce_lib.html>`_
+
+  * ``bounce_health_params``: A dictionary of parameters for get_happy_tasks.
+
+    * ``check_haproxy``: Boolean indicating if PaaSTA should check the local
+      haproxy to make sure this task has been registered and discovered
+      (Defaults to ``True`` if service is in SmartStack)
+
+    * ``min_task_uptime``: Minimum number of seconds that a task must be
+      running before we consider it healthy (Disabled by default)
+
+    * ``haproxy_min_fraction_up``: if ``check_haproxy`` is True, we check haproxy on up to 20 boxes to see whether a task is available.
+      This fraction of boxes must agree that the task is up for the bounce to treat a task as healthy.
+      Defaults to 1.0 -- haproxy on all queried boxes must agree that the task is up.
+
+  * ``bounce_margin_factor``: proportionally increase the number of old instances
+    to be drained when the crossover bounce method is used.
+    0 < bounce_margin_factor <= 1. Defaults to 1 (no influence).
+    This allows bounces to proceed in the face of a percentage of failures.
+    It doesn’t affect any other bounce method but crossover.
+    See `the bounce docs <bouncing.html>`_ for a more detailed description.
+
+  * ``bounce_start_deadline``: a floating point number of seconds to add to the deadline when deployd notices a change
+    to soa-configs or the marked-for-deployment version of an instance.
+    Defaults to 0. (deadline = now)
+    When deployd has a queue of instances to process, it will choose to process instances with a lower deadline first.
+    Set this to a large positive number to allow deployd to process other instances before this one, even if their
+      soa-configs change or mark-for-deployment happened after this one.
+    This setting only affects the first time deployd processes an instance after a change --
+      instances that need to be reprocessed will be reenqueued normally.
+
+  * ``drain_method``: Controls the drain method; see `drain_lib
+    <generated/paasta_tools.drain_lib.html>`_. Defaults to ``noop`` for
+    instances that are not in Smartstack, or ``hacheck`` if they are.
+
+  * ``drain_method_params``: A dictionary of parameters for the specified
+    drain_method. Valid parameters are any of the kwargs defined for the
+    specified bounce_method in `drain_lib <generated/paasta_tools.drain_lib.html>`_.
+
+  * ``cmd``: The command that is executed. If a string, will be wrapped in ``/bin/sh -c``.
+    If a list, will be executed directly as is with no shell parsing.
+
+  * ``args``: An array of docker args if you use the `"entrypoint"
+    <https://docs.docker.com/reference/builder/#entrypoint>`_ functionality.
+
+  * ``monitoring``: See the `monitoring.yaml`_ section for details.
+
+  * ``autoscaling``: See the `autoscaling docs <autoscaling.html>`_ for valid options and how they work
```

Comment:
This would be different for Kubernetes and Marathon. I'll write another one for Kubernetes later. I guess it's OK to just leave it here for now.

Reply:
I'm really hoping they can be the same

Reply:
@solarkennedy they will be the same for existing functions but I plan to add more functionalities that cannot really be supported by marathon autoscaler. Also, things like frequency of autoscaling is different and should be documented. Also, the architecture/algorithm is different and should be written down for those who are interested.